drivers/sdhc/sdhc_renesas_ra.c

```diff
@@ -0,0 +1,743 @@
+		default:
+			ret = -ENOTSUP;
+			goto end;
+		}
+
+		if (priv->bus_width != bus_width) {
+			/* Set the bus width in the SDHI peripheral. */
+			((R_SDHI0_Type *)cfg->regs)->SD_OPTION =
+				SDHI_PRV_SD_OPTION_DEFAULT |
+				(bus_width_reg << SDHI_PRV_SD_OPTION_WIDTH8_BIT);
+			priv->bus_width = bus_width;
+		}
+	}
+
+	if (ios->clock) {
+		if (ios->clock > priv->props.f_max || ios->clock < priv->props.f_min) {
+			LOG_ERR("Proposed clock outside supported host range");
+			return -EINVAL;
+		}
+
+		if (priv->bus_clock != (uint32_t)ios->clock) {
+			fsp_err = r_sdhi_max_clock_rate_set(p_ctrl, ios->clock);
+			ret = err_fsp2zep(fsp_err);
+			if (ret < 0) {
+				goto end;
+			}
+			priv->bus_clock = ios->clock;
+		}
+	}
+
+	if (ios->timing > 0) {
+		/* Set I/O timing */
+		if (priv->timing != ios->timing) {
+			switch (ios->timing) {
+			case SDHC_TIMING_LEGACY:
+			case SDHC_TIMING_HS:
+			case SDHC_TIMING_SDR12:
+			case SDHC_TIMING_SDR25:
+				break;
+			default:
+				LOG_ERR("Timing mode not supported for this device");
+				ret = -ENOTSUP;
+				break;
+			}
+
+			priv->timing = ios->timing;
+		}
+	}
+end:
+
+	return ret;
+}
+
+/*
+ * Get host properties
+ */
+static int sdhc_ra_get_host_props(const struct device *dev, struct sdhc_host_props *props)
+{
+	struct sdhc_ra_priv *priv = dev->data;
+
+	memcpy(props, &priv->props, sizeof(struct sdhc_host_props));
+	return 0;
+}
+
+static int sdhc_ra_init(const struct device *dev)
+{
+	const struct sdhc_ra_config *config = dev->config;
+	struct sdhc_ra_priv *priv = dev->data;
+	fsp_err_t fsp_err;
+	int timeout = SDHI_PRV_ACCESS_TIMEOUT_US;
+	int ret = 0;
+
+	priv->sdmmc_event.transfer_completed = false;
+	k_sem_init(&priv->sdmmc_event.transfer_sem, 1, 1);
+
+	/* Configure dt provided device signals when available */
+	ret = pinctrl_apply_state(config->pcfg, PINCTRL_STATE_DEFAULT);
+	if (ret < 0) {
+		return ret;
+	}
+	if (priv->sdhi_en.port != NULL) {
+		int err = gpio_pin_configure_dt(&priv->sdhi_en, GPIO_OUTPUT_HIGH);
+
+		if (err) {
+			return err;
+		}
+		k_sleep(K_MSEC(50));
+	}
+
+	k_sem_init(&priv->thread_lock, 1, 1);
+	fsp_err = R_SDHI_Open(&priv->sdmmc_ctrl, &priv->fsp_config);
+	ret = err_fsp2zep(fsp_err);
+
+	if (ret < 0) {
+		LOG_INF("R_SDHI_Open error: %d", fsp_err);
+		return ret; /* I/O error*/
+	}
+
+	k_busy_wait(100);
+
+	k_sem_take(&priv->thread_lock, K_USEC(timeout));
+
+	fsp_err = r_sdhi_hw_cfg(&priv->sdmmc_ctrl);
+	ret = err_fsp2zep(fsp_err);
+	if (ret < 0) {
+		LOG_ERR("failed to init sdmmc media");
+		goto end;
+	}
+	priv->bus_width = SDMMC_BUS_WIDTH_1_BIT;
+	priv->timing = SDHC_TIMING_LEGACY;
+	priv->bus_clock = SDMMC_CLOCK_400KHZ;
+
+end:
+	k_sem_give(&priv->thread_lock);
+	return ret;
+}
+
+static const struct sdhc_driver_api sdhc_api = {.reset = sdhc_ra_reset,
+						.request = sdhc_ra_request,
+						.set_io = sdhc_ra_set_io,
+						.get_card_present = sdhc_ra_get_card_present,
+						.card_busy = sdhc_ra_card_busy,
+						.get_host_props = sdhc_ra_get_host_props};
```

Comment:
Add the `DEVICE_API` wrapper and add a trailing comma, so it gets left aligned.

```suggestion
static DEVIC_API(sdhc, sdhc_api) = {
	.reset = sdhc_ra_reset,
	.request = sdhc_ra_request,
	.set_io = sdhc_ra_set_io,
	.get_card_present = sdhc_ra_get_card_present,
	.card_busy = sdhc_ra_card_busy,
	.get_host_props = sdhc_ra_get_host_props,
};
```

Reply:
Updated, thanks